src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { User } from './Types/User';
+import { Todo } from './Types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [input, setInput] = useState('');
+  const [select, setSelect] = useState('0');
+  const [newTodos, setTodos] = useState(todos);
+  const [validation, setValidation] = useState(false);
+
+  const handlerEventOnSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (select === '0' || input === '') {
+      setValidation(true);
+
+      return;
+    }
+
+    const maxNumber = Math.max(...newTodos.map(item => item.id));
+
+    const selectedUser = usersFromServer
+      .find((author: User) => (author.name === select));
+
+    if (!selectedUser) {
+      return;
+    }
+
+    setTodos(prev => [
+      ...prev,
+      {
+        id: maxNumber + 1,
+        userId: selectedUser.id,
+        completed: false,
+        title: input,
+        user: selectedUser,
+      },
+    ]);
+
+    setSelect('0');
+    setInput('');
+    setValidation(false);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handlerEventOnSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">
+            Title:
+            <input
+              id="title"
+              placeholder="Enter a title"
+              type="text"
+              data-cy="titleInput"
+              value={input}
+              onChange={(event) => {
+                setInput(event.target.value);
+              }}
+            />
+          </label>
+          {validation && input === '' ? (
+            <span className="error">Please enter a title</span>
+          ) : null}
```

Comment:
It would be better to create a variable for error and not to use `validation && input === ''`
```suggestion
          {validation && input === '' && (
            <span className="error">Please enter a title</span>
          )}
```